scripts/archive_migration_functions.sh

```diff
@@ -0,0 +1,482 @@
+    for b in "${standardListArr[@]}"; do
+      if [[ "$a" == "$b" ]]; then
+        intersections+=("$a")
+      fi
+    done
+  done
+
+  echo "${intersections[@]}"
+}
+
+is_zipped() {
+  local archive=$1
+
+  local zip
+  zip=$(find "$archive" -maxdepth 1 -type f -name '*.gz' 2>/dev/null)
+  if [[ "$zip" == "" ]]; then
+    echo ""
+  else
+    echo "$zip"
+  fi
+}
+
+unzip_and_keep() {
+  local archive=$1
+
+  local zip
+  zip=$(find "$archive" -maxdepth 1 -type f -name '*.gz')
+  gunzip --keep "$zip"
+}
+
+_guard() {
+  local archive=$1
+  local failures=()
+  local archiveJson="${archive}/data.json"
+
+  if [[ ! -d "$archive" ]]; then
+    failures+=("Archive does not exist")
+  fi
+
+  if [ "${#failures[@]}" -eq 0 ]; then
+    local zipped
+    zipped=$(is_zipped "$archive")
+    if [[ "$zipped" != "" && ! -f "$archiveJson" ]]; then
+      unzip_and_keep "$archive"
+    fi
+
+    local containsDotKibanaRefs
+    containsDotKibanaRefs=$(grep -is '"index": ".kibana' "$archiveJson")
+    if [[ "$containsDotKibanaRefs" == "" ]]; then
+      failures+=(".kibana is not mentioned within the archive")
+    fi
+  fi
+
+  echo "${failures[@]}"
+}
+
+_types() {
+  local archive=${1:-$orig_archive}
+  local archiveJson="${archive}/data.json"
+
+  if [[ -f "${archive}/data.json" ]]; then
+    local typesList
+    typesList=$(jq 'select(.value.source.type | length > 0) | .value.source.type' "$archiveJson")
+    echo "$typesList" | sed 's/^.//;s/.$//' | sort | uniq # The sed command drops the first and last characters.
+  fi
+}
+
+_find_config() {
+  local test_file=${1}
+
+  local current
+  local parent
+  local grandParent
+  local greatGrand
+  current=$(dirname "$test_file")
+  parent=$(dirname "$current")
+  grandParent=$(dirname "$parent")
+  greatGrand=$(dirname "$grandParent")
+
+  local dirs=("$current" "$parent" "$grandParent" "$greatGrand")
+
+  local configs=()
+  for x in "${dirs[@]}"; do
+    local config=$(find "$x" -maxdepth 1 -type f -name 'config.js' -or -name 'config.ts')
+    if [ -n "$config" ]; then
+      configs+=("$config")
+    fi
+  done
+
+  echo "${configs[@]}"
+}
+
+uniqueify() {
+  local xs=("$@")
+  local xss=()
+  for x in $(printf "%s\n" "${xs[@]}" | sort -u); do
+    if [[ -n "$x" ]]; then
+      xss+=("$x")
+    fi
+  done
+  echo "${xss[@]}"
+}
+
+find_configs() {
+  local testFiles=("$@")
+  local configs=()
+  local uniqConfigs=()
+
+  for testFile in "${testFiles[@]}"; do
+    configs+=($(_find_config "$testFile"))
+  done
+
+  uniqConfigs=($(uniqueify "${configs[@]}"))
+  echo "${uniqConfigs[@]}"
+}
+
+usages_list() {
+  local archive=${1:-$orig_archive}
+
+  grep -q "x-pack" <<<"$archive"
+  isInXpack=$?
+
+  local found_usages=()
+
+  if [[ $isInXpack = 0 ]]; then
+    found_usages+=($(find x-pack/test -type f -print0 | xargs -0 grep -n "$archive" | cut -d ':' -f 1 | uniq))
+  else
+    found_usages+=($(find test -type f -print0 | xargs -0 grep -n "$archive" | cut -d ':' -f 1 | uniq))
+  fi
```

Comment:
These two branches are much faster now that they are looking at `test` and `x-pack/test` respectively.
@LeeDr @marius-dr